dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/InMemoryPermissionsImpl.java

```diff
@@ -181,6 +179,34 @@ public boolean canDoAction(User user, Workflow workflow, Role.Action action) {
         }).orElse(false);
     }
 
+    @Override
+    public void selfDestruct(User user) {
+        if (isSharing(user)) {
+            throw new CustomWebApplicationException("The user is sharing at least one workflow and cannot be deleted.",
+                    HttpStatus.SC_BAD_REQUEST);
+        } else {
+            user.getEntries().stream().forEach(e -> {
```

Comment:
Currently you can only [share hosted workflows](https://github.com/ga4gh/dockstore/blob/develop/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java#L880), so I was only thinking of that case when I wrote this, and I think it will work for that case.

But good point about GitHub orgs, since the code that guards it is somewhere else, we shouldn't rely on that here.

That said, I think this is still OK; I'm just deleting the sharing information, not the actual workflows (that's for #1697, which I haven't got to yet). The call to `isSharing` at the beginning of the method makes sure that the workflow is not being shared with anybody else; and only if it isn't, do we remove the map entry for that workflow.